### PR TITLE
[MM-19339] Fix tabbing in modals and multi post-hover-menus

### PR DESCRIPTION
**Pull request by deanwhillier**
#### Summary
The fix for [this ticket](https://mattermost.atlassian.net/browse/MM-18159) broke keyboard navigation in modals, specifically preventing focused elements in modals from visually indicating they have focus. This PR reverts the fix for the original ticket to fix the broken modal navigation and applies a new fix for the original ticket.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-19339 (new ticket)
https://mattermost.atlassian.net/browse/MM-18159 (original ticket)

### CODE REVIEW

File: utils/a11y_controller.js [hunk 1]
```diff
@@ -296,12 +296,6 @@ export default class A11yController {
      * @param {array or boolean} elementPath - array of element's dom branch or boolean to find section/region of element
      */
     nextElement(element, elementPath = false) {
-        if (
-            this.modalIsOpen ||
-            this.popupIsOpen
-        ) {
-            return;
-        }
```
Comment (deanwhillier) [inline utils/a11y_controller.js]: This is the reverted code.

File: utils/a11y_controller.js [hunk 2]
```diff
@@ -419,7 +413,9 @@ export default class A11yController {
         // setup new active element
         this.activeElement = element;
         this.activeElement.addEventListener(A11yCustomEventTypes.UPDATE, this.handleActiveElementUpdate);
-        this.activeElement.dispatchEvent(new Event(A11yCustomEventTypes.ACTIVATE));
+        if (this.activeElement !== this.activeRegion && this.activeElement !== this.activeSection) {
+            this.activeElement.dispatchEvent(new Event(A11yCustomEventTypes.ACTIVATE));
+        }
```
Comment (deanwhillier) [inline utils/a11y_controller.js]: Minor optimization

File: utils/a11y_controller.js [hunk 3]
```diff
@@ -780,7 +781,7 @@ export default class A11yController {
     }
 
     handleFocus = (event) => {
-        if (!this.mouseIsPressed && this.windowIsFocused) {
+        if (this.lastInputEventIsKeyboard && this.windowIsFocused) {
```
Comment (deanwhillier) [inline utils/a11y_controller.js]: This is the fix for the multiple post hover menus. Prevents focus handling if not using the keyboard.

**Comment by lindy65** [comment]
Thanks @deanwhillier 

Tested on Chrome browser 5.16 release branch and desktop app 4.3 RC5 and

1. Tabbing within account settings is fixed and shows focus on element being tabbed to (there is still [this issue](https://mattermost.atlassian.net/browse/MM-19175) being addressed separately.
2. In compact and normal view modes, after escaping from an image preview, only the post with the image attached shows a hover over.

**Pull request merged**